internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -0,0 +1,229 @@
+		}
+	}
+
+	// Write out resolv.conf
+	if err = writeResolvConf(resolvers); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Renew sets up a long running loop to refresh a network interfaces
+// addressing configuration. Currently this only applies to interfaces
+// configured by DHCP.
+func (n *Networkd) Renew(ifaces ...*nic.NetworkInterface) {
+	var wg sync.WaitGroup
+	for _, iface := range ifaces {
+		for _, method := range iface.AddressMethod {
+			if method.TTL() == 0 {
+				continue
+			}
+			wg.Add(1)
+
+			go n.renew(method, iface.Name, iface.Index)
+		}
+	}
+
+	// We dont ever wg.Done
+	// because this should run forever
+	// Probably a better way to do this
+	wg.Wait()
+}
+
+// renew sets up the looping to ensure we keep the addressing information
+// up to date. We attempt to do our first reconfiguration halfway through
+// address TTL. If that fails, we'll continue to attempt to retry every
+// halflife.
+func (n *Networkd) renew(method address.Addressing, name string, index uint32) {
+	renewDuration := method.TTL() / 2
+	for {
+		<-time.After(renewDuration)
+
+		if err := n.configureInterface(method, name, index); err != nil {
+			log.Printf("failed to renew interface address for %s: %v\n", name, err)
+			renewDuration = (renewDuration / 2)
+		} else {
+			renewDuration = method.TTL() / 2
+		}
+	}
+}
+
+// configureInterface handles the actual address discovery mechanism and
+// netlink interaction to configure the interface
+func (n *Networkd) configureInterface(method address.Addressing, name string, index uint32) error {
+	// TODO s/Discover/Something else/
+	// TODO make context more relevant
+	var err error
+	if err = method.Discover(context.Background(), name); err != nil {
+		// Right now this would only happen during dhcp discovery failure
+		log.Printf("Failed to prep %s: %v", name, err)
+		return err
+	}
+
+	// Netlink message generation
+	msg := address.AddressMessage(method, index)
+
+	// Add address if not exist
+	if err = n.AddressAdd(msg); err != nil {
+		// TODO how do we want to handle failures in addressing?
+		log.Printf("Failed to add address %+v to %s: %v", msg, name, err)
+		return err
+	}
+
+	// Set link MTU if we got a response
+	if err = n.setMTU(index, method.MTU()); err != nil {
+		log.Printf("Failed to set mtu %d for %s: %v", method.MTU(), name, err)
+		return err
+	}
+
+	// Add any routes
+	rMsgs := address.RouteMessage(method, index)
+	for _, r := range rMsgs {
+		if err = n.RouteAdd(r); err != nil {
+			// TODO how do we want to handle failures in routing?
+			log.Printf("Failed to add route %+v for %s: %v", r, name, err)
+			continue
+		}
+	}
+
+	return err
+}
+
+// Hostname returns the first hostname found from the addressing methods.
+func (n *Networkd) Hostname(ifaces ...*nic.NetworkInterface) string {
+	for _, iface := range ifaces {
+		for _, method := range iface.AddressMethod {
+			if method.Hostname() != "" {
+				return method.Hostname()
+			}
+		}
+	}
+
+	return ""
+}
+
+/*
```

Comment:
I'd normally say remove commented code, but this one in particular looks like we can turn into some API data?